test/base/src/androidTest/kotlin/io/realm/test/shared/dynamic/DynamicMutableRealmTests.kt

```diff
@@ -79,60 +88,198 @@ class DynamicMutableRealmTests {
     }
 
+
+    // Main points of unmanaged objects:
+    // - Unmanaged object gives ability to construct full graph outside of realm (1)
+    // - Imports can be done on the fly, but creating the top level managed object is maybe one
+    //   step to annoying (2)
+    // - It would be possible to verify setting attributes but it requires access to the
+    //   realmReference. Both options for this is a bit annoying
+    //   - As argument (3) ... just have to pass the dynamicRealm in all the time
+    //   - Using dynamicRealm.create() as factory (4) ... will maybe cause people to think that the
+    //     objects are already managed
+    //
+    // - Requires people to think about managed/unmanaged, but
+    // - Would align with construction import pattern of typed objects
+    // - Drastically simplify embedded API as we can create unmanaged instances and ownership/parent
+    //   will be implicit from the containing
+    // - Would maybe need `EmbeddedDynamicX` classes to make it explictly that we can only
+    //   copyToRealm non-embedded objects just as we are trying to do with RealmObjects
+    //   ... but since there is no compile time linkage to whether a class is embedded or not,
+    //   then it is maybe ok to leave this out ... combining embeddedness with mutability messes up the type system
+    //
+    // #1
+    // DynamicRealmObject.create(
+    //     "Sample",
+    //     mapOf(
+    //         "name" to "HELLO",
+    //         // FIXME If we could accept a map here with `type` implicitly given from
+    //         "child" to DynamicRealmObject.create("Sample")
+    //     )
+    // )
+
+    // DynamicRealmObject.create("type") {
+    //     "" to "Hello",
+    //     "" to "Hello"
+    // }
+
+    // Or with DynamicRealmObject "extension constructor"
+    // DynamicRealmObject(
+    //     "Sample",
+    //     mapOf(
+    //         "name" to "HELLO",
+    //         "child" to DynamicRealmObject("Sample")
+    //     )
+    // )
+    // Alternatively
+    // #2
+    // val obj = dynamicMutableRealm.copyToRealm(DynamicMutableRealm.create("Sample"))
+    // obj.apply {
+    //     set("name", "Hello")
+    //     val child = DynamicRealmObject.create("Sample")
+    //     set("child", child)
+    // }
+    //
+    // dynamicMutableRealm.copyToRealm(obj)
+    //
+    // With Validation
+    // #3
+    // DynamicMutableRealmObject(
+    //     dynamicMutableObject,
+    //     "Sample",
+    //     mapOf(
+    //         "name" to "HELLO",
+    //         "child" to DynamicMutableRealmObject(dynamicMutableObject, "Sample")
+    //     )
+    // )
+    // #4
+    // dynamicMutableRealm.create(
+    //     "Sample",
+    //     mapOf(
+    //         "name" to "HELLO",
+    //         "child" to dynamicMutableObject.create("Sample")
+    //     )
+    // )
+
     // TODO Add variants for each type
     @Test
-    fun createPrimaryKey() {
+    fun copyToRealm_withPrimaryKey() {
```

Comment:
We would need to test for a lot of error cases, like unknown properties, wrong types etc.

Reply:
There are already cases to check that verifies that we throw when object type and property name does exist and if property type and primary key is of wrong type!?